rfc/drafts/mediachain-rfc-3.md

```diff
@@ -0,0 +1,449 @@
+
+`V1 Indexer` will be a minimal proof of concept, while `Future Indexer` represents a set of future enhancements. Key differences are outlined below:
+
+
+   `V1 Indexer`                                      |  `Future Indexer`
+  ---------------------------------------------------|------------------------------------------------------------------------
+   Centralized.                                      | All components distributed across multiple machines.
+   Off-the-shelf components.                         | Purpose-built components.
+   Single perceptual matching goal.                  | Multiple semantic matching intents supported.
+   No supervised training.                           | Custom supervised relevancy / dedupe training for specific task(s) and dataset at hand.
+   No multi-object detection.                        | Multiple objects detectable per media object, via saliency detection / object localization.
+   No explicit user relevancy feedback.              | Accepting explicit user feedback signals.
+   No implicit user behavioral training.             | Training based on implicit user behavior signals.
+   Basic attack mitigation and spam control.         | Sophisticated attack mitigation.
+   Basic or no regression testing.                   | System to detect regressions caused by newly trained models.
+
+
+In summary, `V1 Indexer` will begin as a simple self-hosted system, composed of off-the-shelf open source components, having only a minimal wrapping JSON/REST API to provide abstraction, and using a small importer component for feeding data to and from the blockchain. This version will use simple, unsupervised models of perceptual similarity, based on simple image and textual heuristics.
+
+
+`Future Indexer` can be expanded to include features such as sophisticated neural CNN + attention models, training on supervised training corpuses setup to directly match the system's end-goals, comprehensive scoring and regression testing systems, and sophisticated handling of multi-modality mixed textual and media inputs. Last, a key challenge of `Future Indexer` will be in efficiently scaling the indexes over multiple machines in a P2P environment.
+
+
+## Modes of Operation:
+
+Functions performed by the `V1 Indexer` can be broken down into the following modes of operation:
+
+- **Ingestion** - `Transactors` feed the `Ingestor` with newly created media. The media is semantically indexed for later search and dedupe needs.
+- **Search** -  End-user submits a media file or media ID, a list of most semantically relevant media IDs is returned.
+- **Dedupe** - Determine which media in the index are duplicates / non-duplicates. Submit a subset of these results to the `Transactors`. This process can be initiated periodically to reprocess all media in the search index - for example after each time the dedupe models are considerably re-trained. Or, it can act only on new media from the `Journal`, received via the `Client`.
```

Comment:
hrm, there seems to be some funny threading with my comments -- they ended up out of order in the pull request.